include/proxy/http2/Http2CommonSession.h

```diff
@@ -169,6 +174,21 @@ class Http2CommonSession
   int64_t read_from_early_data   = 0;
   bool cur_frame_from_early_data = false;
 
+  // Counter for received frames
+  std::atomic<uint64_t> _frame_counts_in[HTTP2_FRAME_TYPE_MAX + 1] = {
```

Comment:
Why do this need to be atomic?  h2 connections shouldn't be handled on multiple threads.

Reply:
I think that is true, but we have a lot of SCOPED_MUTEX_LOCK in http2 code and other counters for H2 use atomic too. I don't want any surprise from this PR.

Reply:
Is there an index enum for these values?

Reply:
There is Http2FrameType, and HTTP2_FRAME_TYPE_MAX is the last item. We need one extra space for unknown type.